syncer/main.py

```diff
@@ -54,137 +127,369 @@ def __init__(self, webinterface, connector):
+
+    def handle_output_status(self, event):
+        if self._enabled and event.get('id') in self._output_mapping:
+            local_id = event.get('id')
+            state = {'status': event.get('status').get('on'),
+                     'dimmer': event.get('status').get('value', "")}
 
-    @output_status
-    def output_status(self, status):
-        if self._enabled is True:
+            self._output_mapping[local_id]["state"] = state
+            remote_outputs = self._output_mapping.get(local_id).get('remote_outputs')
             try:
-                on_outputs = set()
-                for entry in status:
-                    if entry[0] in self._output_mapping:
-                        on_outputs.add(entry[0])
-                for output_id in on_outputs - self._previous_outputs:  # Outputs that are turned on
-                    logger.info('Output ON Event: Local-{0}, Remote-{1}'.format(output_id, self._output_mapping[output_id]))
-                    thread = Thread(target=self._call_remote,
-                                    args=('set_output', {'id': self._output_mapping[output_id],
-                                                         'is_on': '1'}))
-                    thread.start()
-                for output_id in self._previous_outputs - on_outputs:  # Outputs that are turned off
-                    logger.info('Output OFF Event: Local-{0}, Remote-{1}'.format(output_id, self._output_mapping[output_id]))
-                    thread = Thread(target=self._call_remote,
-                                    args=('set_output', {'id': self._output_mapping[output_id],
-                                                         'is_on': '0'}))
-                    thread.start()
-                self._previous_outputs = on_outputs
+                for remote_output in remote_outputs:
+                    logger.info(remote_output)
+                    ip = remote_output.get('gw')
+                    remote_id = remote_output.get('remote')
+                    self.update_remote_output_config_and_state(ip=ip, remote_id=remote_id, state=state, state_only=True)
             except Exception as ex:
-                logger.exception('Error processing outputs: {0}'.format(ex))
+                logger.exception(f"Error processing output event {event}: {ex}")
 
-    def _call_remote(self, api_call, params):
+    def _call_remote(self, api_call, gateway, params=None):
         # TODO: If there's an invalid_token error, call self._login() and try this call again
         retries = 0
         while retries < 3:
             try:
-                if self._token is None:
-                    self._login()
-                response = requests.get(self._endpoint.format(api_call),
+                if gateway.get("headers").get("Authorization") is None:
+                    self._login(gateway)
+                response = requests.get(f"{gateway.get('endpoint')}/{api_call}",
```

Comment:
this only supports GET calls. Maybe add support for the other calls as well for future gateway V1 calls.

Reply:
Done